tests/functional/spec/timeout_modal.spec.js

```diff
@@ -0,0 +1,62 @@
+import TimeoutInterstitialPage from "../generated_pages/timeout_modal/timeout-modal-interstitial.page";
+import TimeoutSubmitPage from "../generated_pages/timeout_modal/submit.page";
+import { TimeoutModalPage } from "../base_pages/timeout-modal.page.js";
+import ThankYouPage from "../base_pages/thank-you.page.js";
+
+describe("Timeout Modal", () => {
+  describe("Given I am completing the survey, ", () => {
+    beforeEach(() => {
+      browser.openQuestionnaire("test_timeout_modal.json");
+    });
+    testSet(TimeoutInterstitialPage);
+  });
+});
+
+describe("Timeout Modal Post Submission", () => {
+  describe("Given I am completing the survey and get to post submission page, ", () => {
+    beforeEach(() => {
+      browser.openQuestionnaire("test_timeout_modal.json");
+      $(TimeoutInterstitialPage.submit()).click();
+      $(TimeoutSubmitPage.submit()).click();
+    });
+    testSet(ThankYouPage);
+  });
+});
+
+function testSet(page) {
+  it("When the session timeout is set to 125 seconds, Then it will make the timeout modal with the option to extend the session visible after 65 seconds and it will redirect after session expires", () => {
+    checkTimeoutModal();
+    browser.pause(65000);
+    expect(browser.getUrl()).to.contain("/session-expired");
+    expect($("body").getHTML())
+      .to.include(
+        "Sorry, you need to sign in again",
+        "This is because you have either:",
+        "been inactive for 45 minutes and your session has timed out to protect your information",
+        "followed a link to a page you are not signed in to",
+        "followed a link to a survey that has already been submitted"
+      )
+      .to.not.include("To protect your information, your progress will be saved and you will be signed out in");
+  });
+
+  it("When I click “Continue survey” button of the timeout modal, Then it will extend the session and modal won‘t reappear in the next 15 seconds, no redirect will happen", () => {
+    checkTimeoutModal();
+    $(TimeoutModalPage.submit()).click();
+    expect($(TimeoutModalPage.timer()).getText()).to.equal("");
+    browser.pause(15000);
+    expect(browser.getUrl()).to.contain(page.pageName);
```

Comment:
We will need to do a page refresh to ensure the session has been extended. If modal didn't extend session but disappeared, this would still page. So refresh the page and ensure the page doesn't say `"Sorry, you need to sign in again",`. So for the next time that asserts session is extended.